apps/concierge_site/lib/controllers/auth_controller.ex

```diff
@@ -0,0 +1,105 @@
+defmodule ConciergeSite.AuthController do
+  use ConciergeSite.Web, :controller
+
+  require Logger
+
+  alias AlertProcessor.Model.User
+  alias AlertProcessor.Repo
+  alias ConciergeSite.SessionHelper
+  alias Plug.Conn
+
+  plug(Ueberauth)
+
+  @spec register(Conn.t(), map()) :: Conn.t()
+  def register(conn, _params) do
+    base_uri = System.get_env("KEYCLOAK_BASE_URI")
+    client_id = System.get_env("KEYCLOAK_CLIENT_ID")
+    redirect_uri = "KEYCLOAK_REDIRECT_URI" |> System.get_env() |> URI.encode()
+
+    registration_uri =
+      "#{base_uri}/auth/realms/MBTA/protocol/openid-connect/registrations?client_id=#{client_id}&response_type=code&scope=openid&redirect_uri=#{redirect_uri}"
```

Comment:
**question:** any possible issues here related to not URI-encoding the values, especially `redirect_uri`?

Reply:
Sure seems like no reason not to.